File: backend/dto/opengame.py
from pydantic import BaseModel
import uuid
from datetime import datetime

# /api/opengames
# GET
class OpenGame(BaseModel):
    game_id: uuid.UUID
    user_1_id: uuid.UUID | None
    user_2_id: uuid.UUID | None
    start_time: datetime | None
    board_state: list[uuid.UUID | None] | None = None

class GetOpenGamesResponse(BaseModel):
    games: list[uuid.UUID] | None = None

class PostOpenGameRequest(BaseModel):
    user_id: uuid.UUID

class PostOpenGameResponse(BaseModel):
    success: bool
    game_id: uuid.UUID | None = None

File: frontend/src/dto/opengame.ts
export interface OpenGame {
    game_id: string,
    user_1_id: string,
    user_2_id: string | null
}

export interface GetOpenGamesResponse {
    games: Array<string> | null
}

export interface PostOpenGamesResponse {
    success: boolean,
    game_id: string | null
}